src/Microsoft.SqlTools.ServiceLayer/SchemaCompare/SchemaCompareService.cs

```diff
@@ -231,6 +268,51 @@ await requestContext.SendResult(new ResultStatus()
             }
         }
 
+        /// <summary>
+        /// Handles request for schema compare publish database changes script
+        /// </summary>
+        /// <returns></returns>
+        public async Task HandleSchemaComparePublishProjectChangesRequest(SchemaComparePublishProjectChangesParams parameters, RequestContext<SchemaComparePublishProjectResult> requestContext)
+        {
+            SchemaComparePublishProjectChangesOperation operation = null;
+            try
+            {
+                SchemaComparisonResult compareResult = schemaCompareResults.Value[parameters.OperationId];
+                operation = new SchemaComparePublishProjectChangesOperation(parameters, compareResult);
+
+                TaskMetadata metadata = new()
+                {
+                    TaskOperation = operation,
```

Comment:
can you also update the PublishDatabases one to initialize the metadata like this?